packages/go_router_builder/lib/src/route_config.dart

```diff
@@ -356,43 +408,75 @@ abstract class RouteBaseConfig {
-      if (pathValue.isNull) {
-        throw InvalidGenerationSourceError(
-          'Missing `path` value on annotation.',
-          element: element,
+    switch (typeName) {
+      case 'TypedShellRoute':
+        value = ShellRouteConfig._(
+          routeDataClass: classElement,
+          parent: parent,
+          navigatorKey: _generateNavigatorKeyGetterCode(
+            classElement,
+            keyName: r'$navigatorKey',
+          ),
+          parentNavigatorKey: _generateNavigatorKeyGetterCode(
+            classElement,
+            keyName: r'$parentNavigatorKey',
+          ),
         );
-      }
+        break;
+      case 'TypedStatefulShellRoute':
+        value = StatefulShellRouteConfig._(
+          routeDataClass: classElement,
+          parent: parent,
+          navigatorKey: _generateNavigatorKeyGetterCode(
+            classElement,
+            keyName: r'$navigatorKey',
+          ),
+          parentNavigatorKey: _generateNavigatorKeyGetterCode(
+            classElement,
+            keyName: r'$parentNavigatorKey',
+          ),
+        );
+        break;
+      case 'TypedStatefulShellBranch':
+        value = StatefulShellBranchConfig._(
+          routeDataClass: classElement,
+          parent: parent,
+          navigatorKey: _generateNavigatorKeyGetterCode(
+            classElement,
+            keyName: r'$navigatorKey',
+          ),
+          parentNavigatorKey: _generateNavigatorKeyGetterCode(
+            classElement,
+            keyName: r'$parentNavigatorKey',
+          ),
+        );
+        break;
+      default:
+        final ConstantReader pathValue = reader.read('path');
+        if (pathValue.isNull) {
+          throw InvalidGenerationSourceError(
+            'Missing `path` value on annotation.',
+            element: element,
+          );
+        }
 
-      final ConstantReader nameValue = reader.read('name');
-      value = GoRouteConfig._(
-        path: pathValue.stringValue,
-        name: nameValue.isNull ? null : nameValue.stringValue,
-        routeDataClass: classElement,
-        parent: parent,
-        parentNavigatorKey: _generateNavigatorKeyGetterCode(
-          classElement,
-          keyName: r'$parentNavigatorKey',
-        ),
-      );
+        final ConstantReader nameValue = reader.read('name');
+        value = GoRouteConfig._(
+          path: pathValue.stringValue,
+          name: nameValue.isNull ? null : nameValue.stringValue,
+          routeDataClass: classElement,
+          parent: parent,
+          parentNavigatorKey: _generateNavigatorKeyGetterCode(
+            classElement,
+            keyName: r'$parentNavigatorKey',
+          ),
+        );
     }
 
-    value._children.addAll(reader.read('routes').listValue.map<RouteBaseConfig>(
-        (DartObject e) => RouteBaseConfig._fromAnnotation(
+    value._children.addAll(reader
+        .read(typeName == 'TypedStatefulShellRoute' ? 'branches' : 'routes')
```

Comment:
can this be a getter in the RouteBaseConfig? In general we should try to have less check like this in this method as possible.

Reply:
done